src/common/cockpitchannel.c

```diff
@@ -335,13 +333,9 @@ cockpit_channel_actual_send (CockpitChannel *self,
 
   cockpit_transport_send (self->priv->transport, self->priv->id, payload);
 
-  /* A wraparound of our gint64 size */
+  /* A wraparound of our gint64 size? */
   size = g_bytes_get_size (payload);
-  if (G_MAXINT64 - size < self->priv->out_sequence)
-    {
-      self->priv->out_sequence = 0;
-      self->priv->out_window = CHANNEL_FLOW_WINDOW;
-    }
+  g_return_if_fail (G_MAXINT64 - size > self->priv->out_sequence);
```

Comment:
This is not at all equivalent, so I'll rather let @allisonkarlitskaya review that.

Reply:
It is what @allisonkarlitskaya suggested during review, but we didn't make the change there.

Reply:
Everything inside of the braces is, from a practical standpoint, dead code, because even if a channel were to send a gigabyte every second, it would take several centuries before that code would run....  For this reason I was happier with an assert.